### PR TITLE
boost: Add 1.81.0

### PR DESCRIPTION
**Pull request by sanblch**
- [x] I've read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md).
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

Fixed conandata.yml StrictYML linter errors about url lists and patches missing required fields.

### CODE REVIEW

File: recipes/boost/all/conandata.yml [hunk 1]
```diff
@@ -1,70 +1,48 @@
 sources:
```
Comment (uilianries) [inline recipes/boost/all/conandata.yml:1]: Please, revert and keep those mirrors

Comment (sanblch) [reply recipes/boost/all/conandata.yml:1]: Done

Comment (uilianries) [reply recipes/boost/all/conandata.yml:1]: Thank you!

**Comment by uilianries** [comment]
> ERROR: boost/1.81.0: Error in package_info() method, line 1648
>	ConanException: These libraries were expected to be built, but were not built: {'boost_locale'}

For some reason, Boost locale is missing.

**Comment by sanblch** [comment]
> > ERROR: boost/1.81.0: Error in package_info() method, line 1648
> > ConanException: These libraries were expected to be built, but were not built: {'boost_locale'}
> 
> For some reason, Boost locale is missing.

Boost.Locale supports c++11 only starting from 1.81.0. What should I do? Is it better to exclude Boost.Locale from gcc-5, or set -stdc++11 flag for Boost.Locale when building with gcc-5?

**Review by uilianries — APPROVED** [review]
LGTM

**Review by toge — APPROVED** [review]

**Pull request merged**